salt/modules/dockerng.py

```diff
@@ -1555,11 +1591,16 @@ def _valid_network_mode():  # pylint: disable=unused-variable
                 # Ensure that the user didn't just pass 'container:', because
                 # that would be invalid.
                 return
-            raise SaltInvocationError()
+            else:
+                # just a name assume it is a network
+                log.info(
+                    'Assuming network_mode {0!r} is a network.'.format(
```

Comment:
We should avoid using ``!r`` to quote strings, since it invokes ``repr()``. Thus, if the string is unicode, we get something like this:

```python
>>> 'Assuming network_mode {0!r} is a network.'.format(u'foo')                                                                                         
"Assuming network_mode u'foo' is a network."
```

Changing ``{0!r}`` to ``\'{0}\'`` should be sufficient here.

The usage of the ``!r`` flag has crept into the codebase and used pretty widely, and I'm trying to stamp it out where possible, as it's a blocker for Python 3 support.